typescript/packages/plugins/ironclad/src/ironclad.service.ts

```diff
@@ -0,0 +1,735 @@
+import { LoopDepositParameters, LoopWithdrawParameters } from "./parameters";
+import {
+    BorrowIUSDParameters,
+    RepayIUSDParameters,
+    MonitorPositionParameters,
+} from "./parameters";
+import { CalculateMaxWithdrawableParameters } from "./parameters";
+import { getVaultAddress } from "./vaultAddresses";
+
+interface LoopPosition {
+    borrowedAmounts: string[];
+    totalDeposited: string;
+    totalBorrowed: string;
+}
+
+const LENDING_POOL_ADDRESS = "0xB702cE183b4E1Faa574834715E5D4a6378D0eEd3";
+const PROTOCOL_DATA_PROVIDER_ADDRESS =
+    "0x29563f73De731Ae555093deb795ba4D1E584e42E";
+const IUSD_ADDRESS = "0xA70266C8F8Cf33647dcFEE763961aFf418D9E1E4";
+const BORROWER_ADDRESS = "0x9571873B4Df31D317d4ED4FE4689915A2F3fF7d4";
+const TROVE_MANAGER_ADDRESS = "0x829746b34F624fdB03171AA4cF4D2675B0F2A2e6";
+const HINT_HELPERS_ADDRESS = "0xBdAA7033f0A109A9777ee42a82799642a877Fc4b";
+export class IroncladService {
+    constructor() {}
+
+    @Tool({
+        name: "loop_deposit_ironclad",
+        description:
+            "Perform a looped deposit (recursive borrowing) on Ironclad. Send the amount of the asset (in base units) you want to deposit as the initial amount.",
+    })
+    async loopDeposit(
+        walletClient: EVMWalletClient,
+        parameters: LoopDepositParameters
+    ): Promise<LoopPosition> {
+        try {
+            const position: LoopPosition = {
+                borrowedAmounts: [],
+                totalDeposited: "0",
+                totalBorrowed: "0",
+            };
+
+            const asset = parameters.assetAddress;
+            const allowanceResult = await walletClient.read({
+                address: asset as `0x${string}`,
+                abi: ERC20_ABI,
+                functionName: "allowance",
+                args: [walletClient.getAddress(), LENDING_POOL_ADDRESS],
+            });
+            const allowance = (allowanceResult as { value: bigint }).value;
+
+            if (Number(allowance) < Number(parameters.initialAmount)) {
+                await walletClient.sendTransaction({
+                    to: asset,
+                    abi: ERC20_ABI,
+                    functionName: "approve",
+                    args: [LENDING_POOL_ADDRESS, parameters.initialAmount],
+                });
+            }
+
+            // Initial deposit
+            await walletClient.sendTransaction({
+                to: LENDING_POOL_ADDRESS,
+                abi: LENDING_POOL_ABI,
+                functionName: "deposit",
+                args: [
+                    asset,
+                    parameters.initialAmount,
+                    walletClient.getAddress(),
+                    parameters.referralCode,
+                ],
+            });
+
+            position.totalDeposited = parameters.initialAmount;
+            let currentAmount = parameters.initialAmount;
+
+            // Execute loops
+            for (let i = 0; i < parameters.numLoops; i++) {
```

Comment:
Do the loops need to be done here? Couldn't the LLM call this method X times? I feel this function needs to be broken down in several steps